test/spec/link.spec.jsx

```diff
@@ -2,38 +2,152 @@ import React from 'react';
 import { mount } from 'enzyme';
 
 import { PUSH, REPLACE } from 'src/action-types';
-import Link from 'src/link';
+
+import provideRouter from 'src/provider';
+import {
+  Link as RawLink,
+  PersistentQueryLink as RawPersistentQueryLink
+} from 'src/link';
+
+import bootstrap from './util/bootstrap';
+
+const { storeFactory, history } = bootstrap();
 
 describe('Router link component', () => {
-  it('dispatches a PUSH action with the correct href when clicked', done => {
-    const dispatch = action => {
-      expect(action).to.deep.equal({
-        type: PUSH,
-        payload: {
-          pathname: '/yo'
+  describe('Vanilla link', () => {
+    describe('PUSH', () => {
+      const hrefs = [
+        '/home/messages/a-team?test=ing',
+        {
+          pathname: '/home/messages/a-team?test=ing',
+          query: {
+            test: 'ing'
+          }
+        }
+      ];
+      hrefs.forEach(href =>
+        it('dispatches a PUSH action with the correct href when clicked', done => {
+          const assertion = (state, action) => {
+            if (action.type === PUSH) {
+              const { payload } = action;
+              expect(payload).to.have.property('pathname')
+                .that.contains('/home/messages/a-team');
+              expect(payload).to.have.property('query')
+                .that.deep.equals({ test: 'ing' });
+              done();
```

Comment:
If the `expect` statements above fail, the test will timeout.  This is because an `Error` is thrown, but it is thrown outside the stack frame of the `it` block (its asynchronous).  There are a couple of ways you can resolve this:

1. Wrap all async `expect` statements in a `try...catch` block, and pass any caught errors to `done`.
2. `return Promise.resolve().then(() => // ... my expect statements here`.  If the promise rejects, the test will fail with the rejection reason.
3. Make the test only fake-synchronous.  This isn't always possible or desirable, because often library code is asynchronous and outside of your control.